### PR TITLE
Add CH4 stores in Germany for eGon100RE and adjust grid pipeline capcities (CH4 and H2)

### PR DESCRIPTION
**Pull request by AmeliaNadal**
Will partially solve #779 and #686

## Before merging into `dev`-branch, please make sure that

- [x] the `CHANGELOG.rst` was updated.
- [x] new and adjusted code is formated using `black` and `isort`.
- [x] the `Dataset`-version is updated when existing datasets are adjusted.
- [x] the branch was merged into the
      `continuous-integration/run-everything-over-the-weekend`-branch.
- [x] the workflow is running successful in `test mode`.
- [x] the workflow is running successful in `Everything` mode.

## Remaining tasks

- Insert retroffited H2 storages
- CH4 stores abroad

### CODE REVIEW

**Comment by fwitte** [comment]
@AmeliaNadal: I created the PR #796 based on this branch, since they suit each other quite well. You can just merge that into here (I tested it, both your and my changes did work on a clean run) before merging into dev. I have updated the changelog respectively.

Last thing missing on the storages is then the retroffited H2 storages, I think. For that I will use a function introduced in #790 to calculate the H2 storage capacity, given pressure regime stays the same for hydrogen storage in a retrofitted CH4 storage (pure H2 storage at the same pressure, temperature and volume is only about 30 % of the pure CH4 storage capacity with respect to LHV).

File: src/egon/data/datasets/scenario_parameters/parameters.py [hunk 1]
```diff
@@ -549,7 +549,11 @@ def gas(scenario):
 
         costs = read_csv(2050)
 
-        parameters = {"main_gas_carrier": "H2"}
+        parameters = {
```
[the remark this answers is not in the document]
Comment (AmeliaNadal) [reply src/egon/data/datasets/scenario_parameters/parameters.py:552]: I add some information in scenario_parameters/parameters.py in [5e19a68](https://github.com/openego/eGon-data/pull/793/commits/5e19a68643b60441b79ba49a68111194dd2dbe9b)

[the remark this answers is not in the document]
Comment (AmeliaNadal) [reply src/egon/data/datasets/scenario_parameters/parameters.py:552]: I add some information in scenario_parameters/parameters.py in [5e19a68](https://github.com/openego/eGon-data/pull/793/commits/5e19a68643b60441b79ba49a68111194dd2dbe9b)

File: src/egon/data/datasets/hydrogen_etrago/h2_grid.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
 """The central module containing all code dealing with heat sector in etrago
 """
+from operator import ne
+
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/hydrogen_etrago/h2_grid.py:4]: This imported package seems to be unused

**Review by ClaraBuettner — APPROVED** [review]

**Pull request merged**